src/db/migrate.rs

```diff
@@ -618,6 +618,88 @@ pub fn migrate(version: Option<Version>, conn: &mut Client) -> CratesfyiResult<(
             DROP INDEX github_repos_stars_idx;
             ",
         ),
+        migration!(
+            context,
+            27,
+            // description
+            "Add gitlab handling: creation of the new repositories table which replaces and extend \
+             github_repos",
+            // upgrade query
+            "
+                CREATE TABLE repositories (
+                    id SERIAL PRIMARY KEY,
+                    host VARCHAR NOT NULL,
+                    host_id VARCHAR NOT NULL,
+                    name VARCHAR NOT NULL,
+                    description VARCHAR,
+                    last_commit TIMESTAMPTZ,
+                    stars INT NOT NULL,
+                    forks INT NOT NULL,
+                    issues INT NOT NULL,
+                    updated_at TIMESTAMPTZ NOT NULL,
+                    UNIQUE (host, host_id)
+                );
+
+                ALTER TABLE releases ADD COLUMN repository INTEGER
+                    REFERENCES repositories(id) ON DELETE SET NULL;
+
+                INSERT INTO repositories(host, host_id, name, description, last_commit, stars, forks, issues, updated_at)
+                    SELECT 'github', id, name, description, last_commit, stars, forks, issues, updated_at
```

Comment:
Shouldn't this be a URL?
```suggestion
                    SELECT 'github.com', id, name, description, last_commit, stars, forks, issues, updated_at
```

Reply:
No, it's github. Only gitlab really uses the host.

Reply:
Ok, in fact, using the full host will allow me to remove a lot of duplicated code. Thanks for the suggestion!